tests/test_readers.py

```diff
@@ -179,6 +180,18 @@ def checkExposureFitsReader(self, exposureIn, fileName, dtypesOut):
         reader = ExposureFitsReader(fileName)
         self.assertIn('EXPINFO_V', reader.readMetadata().toDict(), "metadata is automatically versioned")
         reader.readMetadata().remove('EXPINFO_V')
+        # ensure EXTNAMEs can be read and make sense
+        astropyReadFile = astropy.io.fits.open(fileName)
```

Comment:
Please use a context manager here so you don't need to do the explicit close (which would never run if the assert fails).

```python
extnames = {...}
with astropy.io.fits.open(fileName) as astropyReadFile:
    for hdu in astropyReadFile[1:]:
        assert
```